sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -24,15 +25,21 @@
 #define DUAL_CHANNEL 2
 #define QUAD_CHANNEL 4
 #define NAME_SIZE 32
+#define MAXIM_SPEAKER_CODEC_DAI "HiFi"
+#define MAXIM_SPEAKER_DEV_NAME "MX98357A:00"
+
+#define SOF_RT5682_SSP_HP(quirk)		((quirk) & GENMASK(2, 0))
+#define SOF_RT5682_SSP_HP_MASK			(GENMASK(2, 0))
+#define SOF_RT5682_MCLK_EN			BIT(3)
+#define SOF_RT5682_MCLK_24MHZ			BIT(4)
+#define SOF_SPEAKER_CODEC_PRESENT		BIT(5)
+#define SOF_RT5682_SSP_SPK(quirk)		((quirk) & GENMASK(8, 6))
+#define SOF_RT5682_SSP_SPK_MASK			(GENMASK(8, 6))
```

Comment:
It'd be conceptually simpler if you had SSP_CODEC and SSP_AMP. The rt5682 could be connected to a speaker over lineout.

Reply:
Thanks @plbossart for the review and suggestion. Will do as suggest and post the changes.

Reply:
Updated the PR based on your suggestion @plbossart 
@bardliao  can you test this on RVP ?

Reply:
@sathyap-chrome Yes. I will update the test result here.